src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ClaimantTellSomethingElseController.java

```diff
@@ -0,0 +1,147 @@
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.et.common.model.ccd.CCDCallbackResponse;
+import uk.gov.hmcts.et.common.model.ccd.CCDRequest;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.CaseDetails;
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.Helper;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.ClaimantTellSomethingElseService;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.TseService;
+
+import java.util.List;
+
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.YES;
+import static uk.gov.hmcts.ethos.replacement.docmosis.constants.TSEConstants.CLAIMANT_REP_TITLE;
+import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.CallbackRespHelper.getCallbackRespEntityErrors;
+import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.CallbackRespHelper.getCallbackRespEntityNoErrors;
+
+@Slf4j
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/claimantTSE")
+public class ClaimantTellSomethingElseController {
+
+    private final ClaimantTellSomethingElseService claimantTseService;
+    private final TseService tseService;
+
+    /**
+     * Callback endpoint to be called when the event ClaimantTSE is about to start.
+     *
+     * @param ccdRequest the request
+     * @return Callback response entity
+     */
+    @PostMapping(value = "/aboutToStart", consumes = APPLICATION_JSON_VALUE)
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Accessed successfully",
+                content = {
+                    @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = CCDCallbackResponse.class))
+                    }),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> aboutToStartClaimantTSE(
+            @RequestBody CCDRequest ccdRequest) {
+
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        if (Boolean.FALSE.equals(Helper.isRespondentSystemUser(caseData))) {
+            caseData.setClaimantTseRespNotAvailable(YES);
+        }
```

Comment:
are we blocking the claimant rep from making an application if the respondent user is not a system user?
I think the rep should be able to make an application regardless if the respondent user is on the system or not

Reply:
not blocking the claimant rep from making the application. just setting the flag that the respondent is not available